xbmc/video/videosync/VideoSyncDarwin.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ *      Copyright (C) 2015 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "system.h"
+
+#if defined(TARGET_DARWIN)
+#include "utils/log.h"
+#include "VideoSyncDarwin.h"
+#include "utils/MathUtils.h"
+#include "video/VideoReferenceClock.h"
+#include "guilib/GraphicContext.h"
+#include "utils/TimeUtils.h"
+#include "windowing/WindowingFactory.h"
+
+bool CVideoSyncDarwin::Setup(PUPDATECLOCK func)
+{
+  CLog::Log(LOGDEBUG, "CVideoSyncCocoa: setting up Cocoa");
+  bool setupOk = false;
+  
+  //init the vblank timestamp
+  m_LastVBlankTime = CurrentHostCounter();
+  UpdateClock = func;
+  m_abort = false;
+  
+  setupOk = InitDisplayLink();
+  if (setupOk)
+  {
+    g_Windowing.Register(this);
+  }
+  
+  return setupOk;
+}
+
+void CVideoSyncDarwin::Run(volatile bool& stop)
+{
+  //because cocoa has a vblank callback, we just keep sleeping until we're asked to stop the thread
+  while(!stop && !m_abort)
+  {
+    Sleep(1000);
```

Comment:
I would reduce to 100ms. 1 sec is kind of long. a user may observe this delay when stopping playback.